torch_np/_ndarray.py

```diff
@@ -115,28 +119,31 @@ def __str__(self):
     ### comparisons ###
     def __eq__(self, other):
         try:
-            t_other = asarray(other).get
+            return _ufunc_impl.equal(self, asarray(other))
         except RuntimeError:
             # Failed to convert other to array: definitely not equal.
-            # TODO: generalize, delegate to ufuncs
             falsy = torch.full(self.shape, fill_value=False, dtype=bool)
             return asarray(falsy)
-        return asarray(self._tensor == asarray(other).get())
 
     def __neq__(self, other):
-        return asarray(self._tensor != asarray(other).get())
+        try:
+            return _ufunc_impl.not_equal(self, asarray(other))
+        except RuntimeError:
+            # Failed to convert other to array: definitely not equal.
+            falsy = torch.full(self.shape, fill_value=True, dtype=bool)
+            return asarray(falsy)
 
     def __gt__(self, other):
-        return asarray(self._tensor > asarray(other).get())
+        return _ufunc_impl.greater(self, asarray(other))
```

Comment:
Perhaps simpler to just call the functions in `_binary_ufuncs`? This way you could simply do
```python
import _binary_ufuncs as bfun

__gt__ = bfun.greater
...
```

Reply:
With a one-line helper function you could also golf this way all the in-place and reverse dunder methods.

Reply:
Sadly, `_binary_ufuncs` has `from _ndarray import asarray`, so it's an import loop.

Reply:
There has to be a way to untangle this. A (dirty) way of working around this for now would be to do a local import of the module within the class. But more generally, perhaps @rgommers has some ideas on how to work around these issues.

Reply:
Yes, I nearly did  a local import; however this feels almost the same level of dirty, and I'd rather hammer it until it works with a reasonable coverage, and think through a general refactor when it does. A refactor is definitely in order (together with asarray/_from_tensor_and_base rationalization we discussed in a parallel thread, a reasonable solution to asarray imports in dtypes.py etc etc)